avro/src/main/java/com/fasterxml/jackson/dataformat/avro/jsr310/ser/AvroLocalDateTimeSerializer.java

```diff
@@ -0,0 +1,60 @@
+package com.fasterxml.jackson.dataformat.avro.jsr310.ser;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonFormatVisitorWrapper;
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonIntegerFormatVisitor;
+import com.fasterxml.jackson.databind.ser.std.StdScalarSerializer;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+/**
+ * Serializer for {@link LocalDateTime} into long value
+ *
+ * Serialized value represents timestamp in a local timezone, regardless of what specific time zone
+ * is considered local, with a precision of one millisecond from 1 January 1970 00:00:00.000.
+ *
+ * Note: In combination with {@link com.fasterxml.jackson.dataformat.avro.schema.DateTimeVisitor} it aims to produce
+ * Avro schema with type long and logicalType local-timestamp-millis:
+ * {
+ *   "type" : "long",
+ *   "logicalType" : "local-timestamp-millis"
+ * }
+ *
+ * Serialization to string is not supported.
+ */
+public class AvroLocalDateTimeSerializer extends StdScalarSerializer<LocalDateTime> {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final AvroLocalDateTimeSerializer INSTANCE = new AvroLocalDateTimeSerializer();
+
+    protected AvroLocalDateTimeSerializer() {
+        super(LocalDateTime.class);
+    }
+
+    @Override
+    public void serialize(LocalDateTime value, JsonGenerator gen, SerializerProvider provider) throws IOException {
+        /**
+         * Number of milliseconds in a local timezone, regardless of what specific time zone is considered local,
+         * from 1 January 1970 00:00:00.000.
+         */
+        final Instant instant = value.toInstant(ZoneOffset.ofTotalSeconds(0));
```

Comment:
Does this not mean UTC, NOT local time zone? That is fine with me (and actually preferable over "local" anything), but if so comment should indicate that.

Reply:
This is just a way to calculate number of milliseconds from 1/Jan/1970, not considering time zone. 

In Avro documentation, and also in JSR310 **_local_** is synonym to **_without time zone_**.

Events occurring 10 minutes after 1/Jan/1970 represented by `LocalDateTime` (1 January 1970 00:**10**:00.000) in New York or in Paris will both serialize to 10 minutes (precisely to number of milliseconds in 10 minutes).

I will think of better commenting that calculation.

Reply:
So local times are only meaningful if serialized & deserialized in the same time zone context.

Reply:
Yes.
"local" type tells you something happened for example at user's time 16:37 or at 19/Jun/2021 16:37, but you do not know time zone, thus one cannot pinpoint exact moment the event occurred on universal time line but I think that is not purpose of these time types.

Reply:
To me it seems that given that it is necessary to temporarily "bind" local date/time, it should NOT rely on contextual zone at all -- as @dinomite suggest, that would result in reader and writer having to agree on timezone which seems like unnecessary potential failure mode.

Instead we really should just force UTC, unless there are strong reason to do something else.

FWTW, the default time offset for Jackson is actually UTC so for many users there would be no difference.
But ones changing that would get into non-operability abyss.

Reply:
@cowtowncoder 
> To me it seems that given that it is necessary to temporarily "bind" local date/time, it should NOT rely on contextual zone at all - as @dinomite suggest, that would result in reader and writer having to agree on timezone which seems like unnecessary potential failure mode.

Local date/time does not contain time zone, it does not rely on object mapper contextual time zone.
`value.toInstant(ZoneOffset.ofTotalSeconds(0))` is just a method how to calculate to serialize.

Reply:
@MichalFoksa Ok, I think we agree: what I was trying to say is that for purposes of calculation, abstract date+time is handled as if it was in UTC/Z -- 0 offset. Which makes sense.